ibc-core/ics04-channel/types/src/error.rs

```diff
@@ -14,90 +14,64 @@ use crate::commitment::PacketCommitment;
 use crate::timeout::TimeoutTimestamp;
 use crate::Version;
 
+/// Errors that arise from the ICS04 Channel module
 #[derive(Debug, Display)]
 pub enum ChannelError {
     /// decoding error: `{0}`
     Decoding(DecodingError),
-    /// invalid channel id: expected `{expected}`, actual `{actual}`
-    InvalidChannelId { expected: String, actual: String },
+    /// packet acknowledgment for sequence `{0}` already exists
+    DuplicateAcknowledgment(Sequence),
+    /// failed verification: `{0}`
+    FailedVerification(ClientError),
+    /// insufficient packet timeout height: should have `{timeout_height}` > `{chain_height}`
+    InsufficientPacketHeight {
+        chain_height: Height,
+        timeout_height: TimeoutHeight,
+    },
+    /// insufficient packet timestamp: should be greater than chain block timestamp
+    InsufficientPacketTimestamp,
     /// invalid channel state: expected `{expected}`, actual `{actual}`
     InvalidState { expected: String, actual: String },
-    /// invalid channel order type: expected `{expected}`, actual `{actual}`
-    InvalidOrderType { expected: String, actual: String },
     /// invalid connection hops length: expected `{expected}`, actual `{actual}`
     InvalidConnectionHopsLength { expected: u64, actual: u64 },
-    /// invalid counterparty: expected `{expected}`, actual `{actual}`
-    InvalidCounterparty {
-        expected: Counterparty,
-        actual: Counterparty,
-    },
+    /// invalid timeout timestamp: `{0}`
+    InvalidTimeoutTimestamp(TimestampError),
+    /// missing acknowledgment status
+    MissingAcknowledgmentStatus,
     /// missing proof
     MissingProof,
     /// missing proof height
     MissingProofHeight,
```

Comment:
Looking into the places these are utilized, Should we remove them in favor of `DecodingError`?

Reply:
[d550bcd](https://github.com/cosmos/ibc-rs/pull/1343/commits/d550bcd483bd1cc1d33b3544e893d924ebd02f1d)